app/file.go

```diff
@@ -379,7 +379,7 @@ func (a *App) DoUploadFile(now time.Time, rawTeamId string, rawChannelId string,
 		info.ThumbnailPath = pathPrefix + nameWithoutExtension + "_thumb.jpg"
 	}
 
-	if err := a.WriteFile(data, info.Path); err != nil {
+	if info.Size, err = a.WriteFile(file, info.Path); err != nil {
```

Comment:
Do all code paths converge through this file? Otherwise, I fear that some `FileInfo` structs never get their `Size` populated.

Specifically, I'm wondering if the `file_info_test.go` tests actually pass?